### PR TITLE
feat(equipment-type): make EquipmentType as enum usable in TS

### PR DESCRIPTION
**Pull request by TheMaskedTurtle**

### CODE REVIEW

**Comment by sBouzols** [comment]
There is already a work about this in #371 no ?

**Comment by TheMaskedTurtle** [comment]
> There is already a work about this in #371 no ?

@sBouzols  well spotted thank you ! I'm gonna synchronize with @flomillot !

File: src/utils/EquipmentType.ts [hunk 1]
```diff
@@ -44,81 +44,117 @@ export const equipmentStyles = {
...
         tagLabel: 'equipment_search/busTag',
     },
-    SWITCH: {
-        name: 'SWITCH',
+    [EquipmentType.SWITCH]: {
+        name: EquipmentType.SWITCH,
         tagLabel: 'equipment_search/switchTag',
     },
+    [EquipmentType.VSC_CONVERTER_STATION]: undefined,
+    [EquipmentType.LCC_CONVERTER_STATION]: undefined,
```
Comment (flomillot) [inline src/utils/EquipmentType.ts:138]: We can't define them also ?

File: src/utils/EquipmentType.ts [hunk 2]
```diff
@@ -44,81 +44,117 @@ export const equipmentStyles = {
...
+}
+
 export const getEquipmentsInfosForSearchBar = (
-    equipmentsInfos,
-    getNameOrId
+    equipmentsInfos: Equipment[],
+    getNameOrId: (e: Identifiable) => string
 ) => {
-    return equipmentsInfos.flatMap((e) => {
+    return equipmentsInfos.flatMap((e): unknown => {
```
Comment (flomillot) [inline src/utils/EquipmentType.ts:155]: Why `unknown` and not `Equipment` ?

File: src/utils/EquipmentType.ts (hunk 1 above)
Comment (TheMaskedTurtle) [reply src/utils/EquipmentType.ts:138]: I think they're both gathered under the HVDC_CONVERTER_STATION, but as I defined it as a Record<EquipmentType, > I had to define it with all the keys. I don't know where this object is used and not sure the tagLabel will make sense for them.

**Review by flomillot — APPROVED** [review]

**Pull request merged**